bundles/binding/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/PlexBinding.java

```diff
@@ -0,0 +1,318 @@
+			PlexBindingConfig config = plexProvider.getConfig(itemName);
+			if (config != null)
+				setInitialState(config);
+		}		
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void allBindingsChanged(BindingProvider provider) {
+		logger.debug("Plex all bindings changed");
+		if (provider instanceof PlexBindingProvider) {
+			PlexBindingProvider plexProvider = (PlexBindingProvider) provider;		
+			setInitialState(plexProvider);
+		}
+	}
+	
+	/**
+	 * Set initial state for all items for all providers  
+	 */
+	private void setInitialState() {
+		for (PlexBindingProvider provider : providers) 
+			setInitialState(provider);
+	}
+	
+	private void setInitialState(PlexBindingProvider provider) {
+		for (String itemName : provider.getItemNames()) {
+			PlexBindingConfig config = provider.getConfig(itemName);
+			if (config != null)
+				setInitialState(config);
+		}
+	}
+
+	private void setInitialState(PlexBindingConfig config) {
+		if (connector != null) {
+			PlexSession session = connector.getSessionByMachineId(config.getMachineIdentifier());
+			if (session == null) 
+				session = new PlexSession();
+	
+			updateConfigFromSession(config, session);
+		}
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		PlexBindingConfig config = getConfig(itemName);
+		if (!config.isReadOnly()) {
+			try {
+				connector.sendCommand(config, command);
+			} catch (IOException e) {
+				logger.error("Cannot send command {} for item {}", command, itemName, e);
+			}
+		} else {
+			logger.warn("Cannot send command for item {}, property {} is read only", config.getItemName(), config.getProperty());
+		}
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+	}
+		
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		logger.debug("Plex binding updated");
+
+		if (config != null) {
+			PlexConnection connection  = new PlexConnection();
+			Enumeration<String> keys = config.keys();
+			
+			while (keys.hasMoreElements()) {
+				String key = keys.nextElement();
+				if ("service.pid".equals(key)) 
+					continue;
+				
+				String value = ((String) config.get(key)).trim();
+				
+				if ("host".equals(key))
+					connection.setHost(value);
+				else if ("port".equals(key))
+					connection.setPort(Integer.valueOf(value));
+				else if ("refresh".equals(key))
+					if (StringUtils.isNumeric(value))
+						refreshInterval = Integer.valueOf(value);
+					else
+						logger.warn("Non-numeric refresh interval {}, using default {}", value, refreshInterval);
+			}
+			
+			logger.debug("Plex config, server at {}:{}", connection.getHost(), connection.getPort());
+			
+			if (!StringUtils.isEmpty(connection.getHost())) {
+				connector = new PlexConnector(connection, new PlexUpdateReceivedCallback() {
```

Comment:
i'd suggest to extract the code to establish the connection into its own method. This should also help to implement a reconnection feature without duplicating code.